packages/components/src/border-control/styles.ts

```diff
@@ -77,8 +51,10 @@ export const innerWrapper = () => css`
 
 export const wrapperWidth = ( width: CSSProperties[ 'width' ] ) => {
 	return css`
-		width: ${ width };
-		flex: 0 0 auto;
+		${ UnitControlWrapper } {
+			width: ${ width };
+			flex: 0 0 auto;
+		}
```

Comment:
Maybe we could remove this override by:

- Applying the width directly on `UnitControl` via the `__unstableInputWidth` prop
- Applying the flex style on our own div wrapper around the `UnitControl`

Reply:
This PR has been updated to now pass the width to the `UnitControl` via the `__unstableWidth` prop. I found that I still needed the flex style on the unit control wrapper to maintain the border control's desired layout.

I've added some more explanatory comments to the code as well an extra style that fixes the intended default 40/60 ratio between the unit control and slider when no width is set.